src/sync/sync_session.cpp

```diff
@@ -347,11 +361,31 @@ void SyncSession::create_sync_session()
                 case ProtocolError::illegal_realm_path:
                 case ProtocolError::no_such_realm:
                 case ProtocolError::permission_denied:
+                case ProtocolError::bad_client_version:
+                    break;
                 case ProtocolError::bad_server_file_ident:
                 case ProtocolError::bad_client_file_ident:
                 case ProtocolError::bad_server_version:
-                case ProtocolError::bad_client_version:
-                case ProtocolError::diverging_histories:
+                case ProtocolError::diverging_histories: {
+                    // Add a SyncFileActionMetadata marking the Realm as needing to be deleted.
+                    auto full_name = new_path + to_string(std::chrono::duration_cast<std::chrono::seconds>(std::chrono::system_clock::now().time_since_epoch()).count());
```

Comment:
It would be better to put a more human-readable date here, but I don't think `std::chrono` has any calendar APIs.